### ISSUE TITLE
inconsistent with paper

### ISSUE THREAD

**Issue by GqqGGqqG** [IsHYuhi/ST-CGAN_Stacked_Conditional_Generative_Adversarial_Networks#14]
This is a very good reproduction job, but I found that the code is different wiht the paper. The article mentions the use of BCE loss for shadow detection and L1 loss for shadow removal. Is my understanding biased?

https://github.com/IsHYuhi/ST-CGAN_Stacked_Conditional_Generative_Adversarial_Networks/blob/69ee87e9d6b225092f71b89d6b78c05b978af36a/train.py#L224

**Comment by IsHYuhi** [comment]
Thank you for your comment.
Sorry, I don't remember the details as I wrote this more than half a year ago. As you pointed out, in the "Training/Implementation settings" section of the paper, it was mentioned that BCE was used for shadow detection.

Am I correct that you mean the L<sub>data1</sub> loss is going to be replaced by BCE loss?

**Comment by GqqGGqqG** [comment]
> Thank you for your comment.
> Sorry, I don't remember the details as I wrote this more than half a year ago. As you pointed out, in the "Training/Implementation settings" section of the paper, it was mentioned that BCE was used for shadow detection.
> 
> Am I correct that you mean the Ldata1 loss is going to be replaced by BCE loss?

Thx for the reply.
Yeah，this is exactly what i want to say.
But I noticed that different people have different ideas about this loss function when they reproduce it.
I just want to know whether different Loss functions will have a big impact on the results of the model.
L1Loss
https://github.com/jiaruixu/st-cgan/blob/cca13376e84043833e4ae12f2ad0a0e887c244d9/main.py#L313
BCELoss
https://github.com/WHUlwb/stcgan/blob/3bba7a3cfed8a80b65c3d6d67b14dae50524896a/nets.py#L5
“The Binary Cross Entropy (BCE) loss is assigned for the objective of image mask regression and L1 loss is utilized for the shadow-free image reconstruction respectively.”

**Comment by IsHYuhi** [comment]
Thank you for telling this to me. 

Certainly, the difference when using either L1 or BCE is a point of interest.
 It is a simple matter of swapping the loss functions, so please experiment with it. 
Just note that the output layer of G1 is Tanh.

**Comment by GqqGGqqG** [comment]
> Thank you for telling this to me.
> 
> Certainly, the difference when using either L1 or BCE is a point of interest.
> It is a simple matter of swapping the loss functions, so please experiment with it.
> Just note that the output layer of G1 is Tanh.

yeah, I am re-doing the experiment. Whatever, thx for your attention and reply.
